src/core/cgroup.c

```diff
@@ -1279,28 +1279,36 @@ assert_cc(BFQ_WEIGHT(10000) == 1000);
 
 static void set_io_weight(Unit *u, uint64_t weight) {
         char buf[STRLEN("default \n")+DECIMAL_STR_MAX(uint64_t)];
+        uint64_t bfq_weight;
 
         assert(u);
 
         /* FIXME: drop this when distro kernels properly support BFQ through "io.weight"
          * See also: https://github.com/systemd/systemd/pull/13335 and
          * https://github.com/torvalds/linux/commit/65752aef0a407e1ef17ec78a7fc31ba4e0b360f9.
          * The range is 1..1000 apparently, and the default is 100. */
-        xsprintf(buf, "%" PRIu64 "\n", BFQ_WEIGHT(weight));
-        (void) set_attribute_and_warn(u, "io", "io.bfq.weight", buf);
+        bfq_weight = BFQ_WEIGHT(weight);
+        xsprintf(buf, "%" PRIu64 "\n", bfq_weight);
+        if (!set_attribute_and_warn(u, "io", "io.bfq.weight", buf))
```

Comment:
`set_attribute_and_warn()` returns negative errno on failure. And, we do not allow to implicit cast to boolean.
Please do like the following:
```
if (set_attribute_and_warn >= 0)
        log_unit_debug(u, "...");
```